src/main/java/com/feedzai/commons/sql/abstraction/engine/impl/DB2Engine.java

```diff
@@ -425,17 +426,92 @@ protected MappedEntity createPreparedStatementForInserts(final DbEntity entity)
         logger.trace(insertStatement);
         logger.trace(insertReturnStatement);
 
-        PreparedStatement ps, psReturn, psWithAutoInc;
+        PreparedStatement ps = null, psReturn = null, psWithAutoInc = null, psUpsert;
```

Comment:
Why this change if implicitly all these refs already point to null?

Reply:
This has to do with the need of the variables to be initialized as they are before a `try` block and in case of the `IllegalArgumentException` that can be thrown when we do not have PKs in our entities, so the variables can be visible in the `catch` block where this exception is handled. 

Also to have the values assigned to the `ps`, `psReturn` and `psWithAutoInc` so when we try to build the upsert and there is an exception, we already have them initilized and do not need to do it again the `catch` block where the `IllegalArgumentException` is handled.